pkg/solana/txm/pendingtx_test.go

```diff
@@ -1327,3 +1408,163 @@ func TestPendingTxContext_ListAllExpiredBroadcastedTxs(t *testing.T) {
+	}{
+		{
+			name:        "non-existent signature => no reorg",
+			memoryState: txmutils.Broadcasted, // doesn't matter, we'll handle this case specially
+			chainState:  txmutils.Broadcasted,
+			wantReorg:   false,
+		},
+		{
+			name:        "memory=Confirmed, chain=Confirmed => no reorg",
+			memoryState: txmutils.Confirmed,
+			chainState:  txmutils.Confirmed,
+			wantReorg:   false,
+		},
+		{
+			name:        "memory=Confirmed, chain=Processed => reorg",
+			memoryState: txmutils.Confirmed,
+			chainState:  txmutils.Processed,
+			wantReorg:   true,
+		},
+		{
+			name:        "memory=Confirmed, chain=NotFound => reorg",
+			memoryState: txmutils.Confirmed,
+			chainState:  txmutils.NotFound,
+			wantReorg:   true,
+		},
+		{
+			name:        "memory=Processed, chain=Confirmed => no reorg",
+			memoryState: txmutils.Processed,
+			chainState:  txmutils.Confirmed,
+			wantReorg:   false,
+		},
+		{
+			name:        "memory=Processed, chain=Processed => no reorg",
+			memoryState: txmutils.Processed,
+			chainState:  txmutils.Processed,
+			wantReorg:   false,
+		},
+		{
+			name:        "memory=Processed, chain=NotFound => reorg",
+			memoryState: txmutils.Processed,
+			chainState:  txmutils.NotFound,
+			wantReorg:   true,
+		},
+		{
+			name:        "memory=Broadcasted, chain=Confirmed => no reorg",
+			memoryState: txmutils.Broadcasted,
+			chainState:  txmutils.Confirmed,
+			wantReorg:   false,
+		},
+		{
+			name:        "memory=Broadcasted, chain=Processed => no reorg",
+			memoryState: txmutils.Broadcasted,
+			chainState:  txmutils.Processed,
+			wantReorg:   false,
+		},
+		{
+			name:        "memory=Broadcasted, chain=NotFound => no reorg",
+			memoryState: txmutils.Broadcasted,
+			chainState:  txmutils.NotFound,
+			wantReorg:   false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// handle special case
+			if tt.name == "non-existent signature => no reorg" {
+				// don't create any signature in memory
+				txID, hasReorg := txs.IsTxReorged(randomSignature(t), tt.chainState)
+				require.False(t, hasReorg, "expected no reorg for unknown sig")
+				require.Empty(t, txID, "expected empty txID for unknown sig")
+				return
+			}
+
+			// create + set memory state, run IsTxReorged and assert for all other test cases
+			creationTxID, sig := setMemoryState(t, txs, tt.memoryState)
+			returnedTxID, hasReorg := txs.IsTxReorged(sig, tt.chainState)
+			require.Equal(t, creationTxID, returnedTxID, "expected same txID")
+			if tt.wantReorg {
+				require.True(t, hasReorg, "expected reorg for memory=%v, chain=%v", tt.memoryState, tt.chainState)
+			} else {
+				require.False(t, hasReorg, "expected no reorg for memory=%v, chain=%v", tt.memoryState, tt.chainState)
+			}
+		})
+	}
+}
+
+func TestPendingTxContext_GetPendingTx(t *testing.T) {
+	t.Parallel()
+	txs := newPendingTxContext()
+
+	t.Run("successfully retrieve broadcasted transaction", func(t *testing.T) {
```

Comment:
nit: based on the code I think the description should be
```suggestion
	t.Run("successfully retrieve processed transaction", func(t *testing.T) {
```

Reply:
Might also be worth adding a test case for Broadcasted too though?

Reply:
good catch, adding it